File: core/src/main/java/pcb/uwu/units/derived/optics/IlluminanceUnit.java
package pcb.uwu.units.derived.optics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.LengthUnit;
import pcb.uwu.units.base.LuminousIntensityUnit;
import pcb.uwu.units.derived.fundamental.AreaUnit;
import pcb.uwu.units.derived.scalar.SolidAngleUnit;

public class IlluminanceUnit extends CompositeUnit {

	// region base constructor

	public IlluminanceUnit(LuminousIntensityUnit luminousIntensityUnit, SolidAngleUnit solidAngleUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(luminousIntensityUnit.getUnitCounter())
				.major(solidAngleUnit)
				.minor(lengthUnit, 2));
	}

	public IlluminanceUnit(LuminousFluxUnit luminousFluxUnit, AreaUnit areaUnit) {
		super(new UnitCounter()
				.major(luminousFluxUnit.getUnitCounter())
				.minor(areaUnit.getUnitCounter()));
	}

	// endregion

}

File: dimension/src/main/java/pcb/uwu/unit/derived/optics/IlluminanceUnit.kt
package pcb.uwu.unit.derived.optics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.LuminousIntensityUnit
import pcb.uwu.unit.derived.geometry.AreaUnit
import pcb.uwu.unit.derived.geometry.SolidAngleUnit

open class IlluminanceUnit : CompositeUnit
{
    constructor(luminousIntensityUnit: LuminousIntensityUnit,
                solidAngleUnit: SolidAngleUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(luminousIntensityUnit)
                        .major(solidAngleUnit)
                        .minor(lengthUnit)
                        .minor(lengthUnit))

    constructor(luminousFluxUnit: LuminousFluxUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(luminousFluxUnit)
                        .minor(areaUnit))
}
